File: lib/src/main/java/one/hyro/spark/lib/common/Rank.java
package one.hyro.spark.lib.common;

import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.format.NamedTextColor;

public enum Rank {
    OWNER(Component.text("OWNER")),
    ADMIN(Component.text("ADMIN")),
    MODERATOR(Component.text("MODERATOR")),
    HELPER(Component.text("HELPER")),
    SPARK_PLUS_PLUS(Component.text("SPARK++")),
    SPARK_PLUS(Component.text("SPARK+")),
    SPARK(Component.text("SPARK")),
    DEFAULT(Component.text("DEFAULT"));

    private final Component prefix;

    Rank(Component prefix) {
        this.prefix = prefix;
    }

    public Component getPrefix() {
        return Component.text("[", NamedTextColor.GRAY)
                .append(prefix)
                .append(Component.text("]", NamedTextColor.GRAY))
                .append(Component.space());
    }
}

File: lib/src/main/kotlin/common/Rank.kt
package one.hyro.common

import net.kyori.adventure.text.Component
import net.kyori.adventure.text.format.NamedTextColor

enum class Rank(private val prefix: Component?, private val weight: UByte, val permissions: List<String>) {
    USER(null, 0u, listOf()),
    HELPER(
        Component.text("[", NamedTextColor.DARK_GRAY)
            .append(Component.text("HELPER", NamedTextColor.DARK_GREEN)
                .append(Component.text("]", NamedTextColor.DARK_GRAY))
            ),
        1u,
        listOf()
    ),
    MODERATOR(
        Component.text("[", NamedTextColor.DARK_GRAY)
            .append(Component.text("MOD", NamedTextColor.BLUE)
                .append(Component.text("]", NamedTextColor.DARK_GRAY))
            ),
        2u,
        listOf()
    ),
    ADMINISTRATOR(
        Component.text("[", NamedTextColor.DARK_GRAY)
            .append(Component.text("ADMIN", NamedTextColor.RED)
                .append(Component.text("]", NamedTextColor.DARK_GRAY))
            ),
        3u,
        listOf()
    ),
    OWNER(
        Component.text("[", NamedTextColor.DARK_GRAY)
            .append(Component.text("OWNER", NamedTextColor.GOLD)
                .append(Component.text("]", NamedTextColor.DARK_GRAY))
            ),
        4u,
        listOf()
    );

    fun prefix(): Component = prefix?.appendSpace() ?: Component.empty()
    //fun weightname(): String = weight.toString() + name.lowercase()
}